keychain-sdk/writer.go

```diff
@@ -51,17 +47,17 @@ func NewTxWriter(
 
 func (w *TxWriter) Start(ctx context.Context, flushErrors chan error) error {
 	w.Logger.Info("starting tx writer")
-	ticker := time.NewTicker(w.BatchTimeout)
-	defer ticker.Stop()
-
 	for {
 		select {
-		case <-ticker.C:
+		case <-ctx.Done():
+			return nil
+		default:
+			ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
```

Comment:
I think we should have the  timeout as a configurable param? WDYT?

Reply:
If not, I'd put it as `const`

Reply:
I renamed "BatchTimeout" into "BatchInterval" and added the new param called "TxTimeout"

(we'll also need to update the env var for the deployed services)